unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarModifierAreaHUD/Scripts/AvatarModifierAreaFeedbackView.cs

```diff
@@ -0,0 +1,206 @@
+            if (setVisible)
+            {
+                Show();
+            }
+            else
+            {
+                Hide();
+            }
+        }
+        
+        public override void OnFocus()
+        {
+            base.OnFocus();
+            
+            if (!isVisible) return;
+
+            ShowWarningMessage();
+            HideIcon();
+        }
+
+        public override void OnLoseFocus()
+        {
+            base.OnLoseFocus();
+
+            if (!isVisible) return;
+            
+            HideWarningMessage();
+            ShowIcon();
+        }
+        
+        private async UniTaskVoid HideFirstTimeWarningMessage(CancellationToken cancellationToken)
+        {
+            await UniTask.Delay(5000, cancellationToken: cancellationToken);
+            await UniTask.SwitchToMainThread(cancellationToken);
+            if (cancellationToken.IsCancellationRequested) return;
+            HideWarningMessage();
+            ShowIcon();
+        }
+
+        private void HideWarningMessage()
+        {
+            if (warningMessageAnimationCoroutine != null)
+            {
+                StopCoroutine(warningMessageAnimationCoroutine);
+            }
+            warningMessageAnimationCoroutine = StartCoroutine(WarningMessageAnimationCoroutine(Vector3.zero, 0));
+        }
+        
+        private void ShowWarningMessage()
+        {
+            if (warningMessageAnimationCoroutine != null)
+            {
+                StopCoroutine(warningMessageAnimationCoroutine);
+            }
+            warningMessageAnimationCoroutine = StartCoroutine(WarningMessageAnimationCoroutine(Vector3.one, 1));
+            currentState = AvatarModifierAreaFeedbackState.WARNING_MESSAGE_VISIBLE;
+        }
+        
+        private void HideIcon()
+        {
+            if (iconAnimationCoroutine != null)
+            {
+                StopCoroutine(iconAnimationCoroutine);
+            }
+            iconAnimationCoroutine = StartCoroutine(IconAnimationCoroutine(0));
+        }
+        
+        private void ShowIcon()
+        {
+            if (iconAnimationCoroutine != null)
+            {
+                StopCoroutine(iconAnimationCoroutine);
+            }
+            
+            warningIconCanvasGroup.blocksRaycasts = true;
+            
+            iconAnimationCoroutine = StartCoroutine(IconAnimationCoroutine(1));
+            currentState = AvatarModifierAreaFeedbackState.ICON_VISIBLE;
+        }
+
+        public void AddNewWarningMessage(string newWarning)
+        {
+            if (!string.IsNullOrEmpty(descriptionText.text)) descriptionText.text += "\n";
+                
+            descriptionText.text += newWarning;
+        }
+
+        public void ResetWarningMessage()
+        {
+            descriptionText.text = "";
+        }
+
+        public override void RefreshControl() { }
+
+        IEnumerator WarningMessageAnimationCoroutine(Vector3 destinationScale, float destinationAlpha)
```

Comment:
I would have UniTasks instead of coroutines. They can be tricky at first (mostly because cancelling them is a bit convoluted) but it lead to more testable/clean code.